components/post_view/post_header/post_header.tsx

```diff
@@ -1,96 +1,97 @@
 
-export default class PostHeader extends React.PureComponent {
-    static propTypes = {
-
-        /*
-         * The post to render the header for
-         */
-        post: PropTypes.object.isRequired,
-
-        /*
-         * Function called when the comment icon is clicked
-         */
-        handleCommentClick: PropTypes.func.isRequired,
-
-        /*
-         * Function called when the card icon is clicked
-         */
-        handleCardClick: PropTypes.func.isRequired,
-
-        /*
-         * Function called when the post options dropdown is opened
-         */
-        handleDropdownOpened: PropTypes.func.isRequired,
-
-        /*
-         * Set to render compactly
-         */
-        compactDisplay: PropTypes.bool,
-
-        /*
-         * The number of replies in the same thread as this post
-         */
-        replyCount: PropTypes.number,
-
-        /**
-         * Set to indicate that this is previous post was not a reply to the same thread
-         */
-        isFirstReply: PropTypes.bool,
-
-        /**
-         * Set to mark post as being hovered over
-         */
-        hover: PropTypes.bool.isRequired,
-
-        /*
-         * Set to render the post time when not hovering
-         */
-        showTimeWithoutHover: PropTypes.bool,
-
-        /**
-         * Whether or not the post username can be overridden.
-         */
-        enablePostUsernameOverride: PropTypes.bool.isRequired,
-
-        /**
-         * If the user that made the post is a bot.
-         */
-        isBot: PropTypes.bool.isRequired,
-
-        /**
-         * If the user that made the post is a guest.
-         */
-        isGuest: PropTypes.bool.isRequired,
-
-        /**
-         * To Check if the current post is last in the list
-         */
-        isLastPost: PropTypes.bool,
-
-        /**
-         * Source of image that should be override current user profile.
-         */
-        overwriteIcon: PropTypes.string,
-    }
-
-    render() {
+export type Props = {
+
+    /*
+    * The post to render the header for
+    */
+    post: Post;
+
+    /*
+    * Function called when the comment icon is clicked
+    */
+    handleCommentClick: EventHandler<MouseEvent>;
+
+    /*
+    * Function called when the card icon is clicked
+    */
+    handleCardClick: (post: Post) => void;
+
+    /*
+    * Function called when the post options dropdown is opened
+    */
+    handleDropdownOpened: (opened: boolean) => void;
+
+    /*
+    * Set to render compactly
+    */
+    compactDisplay?: boolean;
+
+    /*
+    * The number of replies in the same thread as this post
+    */
+    replyCount?: number;
+
+    /**
+     * Set to indicate that this is previous post was not a reply to the same thread
+     */
+    isFirstReply?: boolean;
+
+    /**
+     * Set to mark post as being hovered over
+     */
+    hover: boolean;
+
+    /*
+    * Set to render the post time when not hovering
+    */
+    showTimeWithoutHover: boolean;
```

Comment:
This was originally optional. However, I've traced it down in the whole project, and found that it will always have a Boolean value. So, I think making it required is better than checking for `undefined` in the code (if left optional) or using `as Boolean`

Reply:
@abdusabri thanks! I agree that if it always has a value it will be best to have it as required.